infra/modules/aurora-db/main.tf

```diff
@@ -0,0 +1,201 @@
+data "aws_caller_identity" "current" {}
+data "aws_region" "current" {}
+
+
+###########################
+## Database Configuration ##
+###########################
+resource "aws_rds_cluster" "postgresql" {
+  # checkov:skip=CKV2_AWS_27:have concerns about sensitive data in logs; want better way to get this information
+  # checkov:skip=CKV2_AWS_8:TODO add backup selection plan using tags
+  cluster_identifier = var.service_name
+  engine             = "aurora-postgresql"
+  engine_mode        = "provisioned"
+  database_name      = replace("${var.service_name}", "-", "_")
+  master_username    = "app_usr"
+  master_password    = aws_ssm_parameter.random_db_password.value
+  storage_encrypted  = true
+  # checkov:skip=CKV_AWS_128:Auth decision needs to be ironed out
+  # checkov:skip=CKV_AWS_162:Auth decision needs to be ironed out
+  # iam_database_authentication_enabled = true
+  deletion_protection = true
+  # final_snapshot_identifier = "${var.service_name}-final"
+  skip_final_snapshot = true
+
+
+  serverlessv2_scaling_configuration {
+    max_capacity = 1.0
+    min_capacity = 0.5
+  }
+
+}
+
+resource "aws_rds_cluster_instance" "postgresql-cluster" {
+  cluster_identifier         = aws_rds_cluster.postgresql.id
+  instance_class             = "db.serverless"
+  engine                     = aws_rds_cluster.postgresql.engine
+  engine_version             = aws_rds_cluster.postgresql.engine_version
+  auto_minor_version_upgrade = true
+  monitoring_role_arn        = aws_iam_role.rds_enhanced_monitoring.arn
+  monitoring_interval        = 30
+}
+
+resource "random_password" "random_db_password" {
+  length           = 48
+  special          = true
+  min_special      = 6
+  override_special = "!#$%&*()-_=+[]{}<>:?"
```

Comment:
as far as i know there aren't any constraints on aurora serverless master passwords so we don't need to force special characters, since there isn't any added security benefit and we can keep this simpler / reduce lines of code by getting rid of that config